pkg/controller/controller.go

```diff
@@ -533,11 +551,34 @@ func (ctrl *resizeController) patchClaim(oldPVC, newPVC *v1.PersistentVolumeClai
 	return updatedClaim, nil
 }
 
-func (ctrl *resizeController) updatePVCapacity(pv *v1.PersistentVolume, newCapacity resource.Quantity) error {
+func (ctrl *resizeController) deletePreResizeCapAnnotation(pv *v1.PersistentVolume) error {
+	// if the pv does not have a resize annotation skip the entire process
+	if !metav1.HasAnnotation(pv.ObjectMeta, util.AnnPreResizeCapacity) {
+		return nil
+	}
+	pvClone := pv.DeepCopy()
+	delete(pvClone.ObjectMeta.Annotations, util.AnnPreResizeCapacity)
+
+	_, err := ctrl.patchPersistentVolume(pv, pvClone)
+	return err
+}
+
+func (ctrl *resizeController) updatePVCapacity(pv *v1.PersistentVolume, oldCapacity, newCapacity resource.Quantity, fsResizeRequired bool) error {
 	klog.V(4).Infof("Resize volume succeeded for volume %q, start to update PV's capacity", pv.Name)
 	newPV := pv.DeepCopy()
 	newPV.Spec.Capacity[v1.ResourceStorage] = newCapacity
 
+	if fsResizeRequired {
+		// if the pv already has a resize annotation skip
+		if metav1.HasAnnotation(pv.ObjectMeta, util.AnnPreResizeCapacity) {
+			return nil
+		}
+		if newPV.ObjectMeta.Annotations == nil {
+			newPV.ObjectMeta.Annotations = make(map[string]string)
+		}
+		newPV.ObjectMeta.Annotations[util.AnnPreResizeCapacity] = (&oldCapacity).String()
```

Comment:
Why are we doing - `(&oldCapacity).String()` is that different from `oldCapacity.String()` ?

Reply:
Done :)